src/discord_integration.py

```diff
@@ -61,69 +62,44 @@ def validate_token() -> bool:
         os.remove(platformdirs.user_config_dir("Qtcord") + "/discordauth.txt")
 
 
-def get_messages(channel_id: int, limit: int = 100) -> dict:
+def get_messages(channel_id: int, limit: int = 100) -> dict[int, DiscordMessage]:
     """
     Retrives messages from a specified channel.
 
     Args:
-        channel_id (int): The channel ID to request from
+        channel_id (int): A channel ID.
         limit (int, optional): The maximum messages to request. Default is 100. May not go higher.
 
     Returns:
-        dict: The channel ID as the only key, and a list with the messages from the channel as its value.
+        dict[int, DiscordMessage]: A dictionary with the channel ID as only key, and a list with
+          the channel's messages as its value.
     """
 
     r = requests.get(
         f"{api_base}/channels/{channel_id}/messages?limit={limit}", headers=headers
     )
 
-    new_list = []
+    messages_list = []
 
     if r.status_code != 200:
-        new_list.append(
-            {
-                "timestamp": datetime.now(),
-                "username": "System",
-                "content": "Error. This may be a forum channel, or you're not allowed to view the content.",
-                "id": 0,
-            }
+        messages_list.append(
+            DiscordMessage(
+                id=0,
+                author=DiscordUser(0, "", "", "System"),
+                content="Error. This may be a forum channel, or you're not allowed to view the content.",
+                attachments=[],
+                timestamp=datetime.utcfromtimestamp(0),
+                referenced_message=None,
+            )
         )
-        return {channel_id: new_list}
+        return {channel_id: messages_list}
```

Comment:
This is throwing off squiggly lines in my IDE because it's not returning a DiscordMessage. Should I just remove the "DiscordMessage" part from this line and replace it with `list`? `def get_messages(channel_id: int, limit: int = 100) -> dict[int, DiscordMessage]:`

Reply:
Never mind, I'll just do this. Object if needed.